### PR TITLE
Support JDK 20 (compiling and testing only)

### LINKED ISSUE #23488: Support running Gradle on Java 20

### LINKED ISSUE #23489: Support compiling and testing with Java 20

### PR DESCRIPTION
**Pull request by octylFractal**
Fixes #23489.

This includes minor changes to allow us to compile, test, etc. Java and Groovy on JDK 20.

### CODE REVIEW

**Comment by blindpirate** [comment]
Note that you need to push to `experimental` branch to test it.

**Comment by octylFractal** [comment]
Are you sure? It seems to pick up my configuration changes just fine, though it seems we're missing JDK 20 on Windows (maybe located at `open-jdk-20-ea` instead?): https://builds.gradle.org/buildConfiguration/Gradle_Master_Check_Platform_4_bucket22/61685902?hideProblemsFromDependencies=false&hideTestsFromDependencies=false&expandBuildProblemsSection=true&expandBuildChangesSection=true

**Comment by blindpirate** [comment]
@octylFractal You are right, you can test it on `master` pipeline. However, I found some tricky thing: the JDK 20 we've installed is actually "OpenJDK, Oracle Build", which has a vendor `oracle` instead of `openjdk`. So as a workaround, I changed the vendor to `oracle`: https://github.com/gradle/gradle/commit/944c2b3e910c645ed6fe239daa03a59558e18c32
and added necessary `%linux.java20.oracle.64bit%`, `%windows.java20.oracle.64bit%`. Now the build is running:

https://builds.gradle.org/buildConfiguration/Gradle_Master_Check_Stage_QuickFeedback_Trigger/61712841?hideProblemsFromDependencies=false&hideTestsFromDependencies=true&expandBuildTestsSection=true&expandBuildChangesSection=true

**Comment by octylFractal** [comment]
@bot-gradle test this

File: subprojects/docs/src/docs/release/notes.md [hunk 1]
```diff
@@ -87,6 +87,12 @@ TODO - Java lambdas are supported, and unsupported captured values are reported.
 TODO - File collections queried at configuration time are treated as configuration inputs.
 TODO - File system repositories are fully supported including dynamic versions in Maven, Maven local, and Ivy repositories
 
+### JVM
+
+#### Added Support for using Java 20 via Toolchains
+
+Gradle now supports using Java 20 via [Toolchains](userguide/toolchains.html). Running Gradle on Java 20 is not yet supported.
```
Comment (hythloda) [inline subprojects/docs/src/docs/release/notes.md:94]: The guidelines for this are:

- Set the context - what are we even talking about (don’t assume too much about what the user already knows)
- Express the problem or use case in the given context
- Explain how the new release addresses that problem or use case
- Provide build script snippet or screenshot illustrating the new feature, if applicable
- Link to the full documentation for more details 

Anything you can add to this for it to be more relatable?

File: subprojects/docs/src/docs/release/notes.md [hunk 2]
```diff
@@ -113,6 +113,12 @@ TODO - Java lambdas are supported, and unsupported captured values are reported.
...
```
Comment (big-guy) [inline subprojects/docs/src/docs/release/notes.md:120]:
```suggestion
#### Support for compiling and testing with Java 20

Gradle now supports compiling, testing and running projects using Java 20.

Builds can be configured to use a Java 20 [toolchain](userguide/toolchains.html); however, running Gradle on Java 20 is not yet supported.
```

**Review by big-guy — APPROVED** [review]
Suggestion for the release notes, but everything is OK once we fix the problem with the distribution tests.

**Review by big-guy — APPROVED** [review]

File: subprojects/docs/src/docs/release/notes.md (hunk 1 above)
Comment (octylFractal) [reply subprojects/docs/src/docs/release/notes.md]: I moved this to the 8.1 release notes Google doc.

**Comment by octylFractal** [comment]
@bot-gradle test this

**Comment by big-guy** [comment]
@bot-gradle test and merge

**Comment by octylFractal** [comment]
@bot-gradle test and merge

**Pull request merged**